File: frontend/src/lib/writingIntent.ts
/** 写作意图（与 backend/config/writing_intents.yaml 对齐） */

export type WritingIntentId =
  | 'informational'
  | 'insight_commentary'
  | 'literary_essay';

export type WritingIntentOption = {
  id: WritingIntentId;
  label: string;
  description?: string;
};

export const FALLBACK_WRITING_INTENTS: WritingIntentOption[] = [
  {
    id: 'insight_commentary',
    label: '观点评论',
    description: '有论断、证据链与判断句',
  },
  {
    id: 'informational',
    label: '信息科普',
    description: '覆盖要点、结构清晰',
  },
  {
    id: 'literary_essay',
    label: '叙事随笔',
    description: '适度修辞与作者视角',
  },
];

export const FALLBACK_DEFAULT_INTENT: WritingIntentId = 'insight_commentary';

export function normalizeWritingIntentId(
  value: string | null | undefined,
): WritingIntentId {
  if (
    value === 'informational' ||
    value === 'insight_commentary' ||
    value === 'literary_essay'
  ) {
    return value;
  }
  return FALLBACK_DEFAULT_INTENT;
}

File: backend/app/prompts/writing_intent.py
"""写作意图（config/writing_intents.yaml）"""
from __future__ import annotations

from ..config_loader import get_writing_intent_catalog

ORNAMENTAL_INTENTS = frozenset({'literary_essay'})


def normalize_writing_intent(value: str | None) -> str:
    catalog = get_writing_intent_catalog()
    intents = catalog.get('intents') or {}
    default = str(catalog.get('default') or 'insight_commentary')
    if value and value in intents:
        return value
    return default


def get_writing_intent_meta(intent_id: str) -> dict[str, str] | None:
    intents = get_writing_intent_catalog().get('intents') or {}
    meta = intents.get(intent_id)
    if not meta or not isinstance(meta, dict):
        return None
    return {
        'id': intent_id,
        'label': str(meta.get('label') or intent_id),
        'description': str(meta.get('description') or ''),
    }


def allows_literary_rhetoric(intent_id: str) -> bool:
    return intent_id in ORNAMENTAL_INTENTS


def parse_writing_intent_from_data(data: dict | None) -> str:
    if not data:
        return normalize_writing_intent(None)
    raw = data.get('writing_intent') or data.get('writingIntent')
    if not raw and isinstance(data.get('context'), dict):
        ctx = data['context']
        raw = ctx.get('writing_intent') or ctx.get('writingIntent')
    return normalize_writing_intent(str(raw).strip() if raw else None)
